hledger-lib/Hledger/Data/Types.hs

```diff
@@ -230,7 +230,8 @@ data Amount = Amount {
       aprice      :: !(Maybe AmountPrice)  -- ^ the (fixed, transaction-specific) price for this amount, if any
     } deriving (Eq,Ord,Generic,Show)
 
-newtype MixedAmount = Mixed [Amount] deriving (Eq,Ord,Generic,Show)
+newtype MixedAmount = Mixed (M.Map MixedAmountKey Amount) deriving (Eq,Ord,Generic,Show)
+type MixedAmountKey = (CommoditySymbol, Maybe (CommoditySymbol, Maybe Quantity))
```

Comment:
Any concerns about using a tuple here rather than an ADT, since they often seem to get blamed for space leaks/excessive laziness/thunk buildup ?

Reply:
It's probably better to use and ADT, I was just copying over the same data representation as used in the previous normalisation command.